tests/test_cases/test_discovery/test_discovery.py

```diff
@@ -471,3 +471,17 @@ def type_check_verilog(dut):
         tlog.info("Handle %s" %  (handle[0]._fullname,))
         if handle[0]._type != handle[1]:
             raise TestFailure("Expected %s found %s for %s" % (handle[1], handle[0]._type, handle[0]._fullname))
+
+
+@cocotb.test(skip=cocotb.LANGUAGE in ["vhdl"])
+def test_access_underscore_name(dut):
+    """Test accessing HDL name starting with an underscore"""
+    dut._id("_underscore_name", extended=False) <= 0
+    yield Timer(1, 'ns')
+    assert dut._id("_underscore_name", extended=False).value == 0
+    dut._id("_underscore_name", extended=False) <= 1
+    yield Timer(1, 'ns')
+    assert dut._id("_underscore_name", extended=False).value == 1
+    dut._id("_underscore_name", extended=False) <= 0
+    yield Timer(1, 'ns')
+    assert dut._id("_underscore_name", extended=False).value == 0
```

Comment:
What does `dut._underscore_name` give here? I'd argue it should either fail or emit a deprecation warning

Reply:
For some reason, it just works for me now, but I'm sure it failed before.

Reply:
Perhaps I accidentally changed it in #1510.

Reply:
Yep, I did - ok to add those lines back in this PR, and add the test for the expected `AttributeError`?

Reply:
Okay for me, but it may take me a few days until I can work on this again. Feel free to adopt.

Reply:
Like this?